libkmod/libkmod-config.c

```diff
@@ -794,7 +794,7 @@ static int conf_files_insert_sorted(struct kmod_ctx *ctx,
 	bool is_single = false;
 
 	if (name == NULL) {
-		name = basename(path);
```

Comment:
general approach in kmod is to consider the gnu-behavior as the desired one and add missing stuff for other libc's where appropriate. That way we don't have to keep patching kmod when the next .c uses basename() rather than gnu_basename().

Reply:
I understand that the GNU basename() behaviour is more desirable (and agree that it is), but the unfortunate fact is that the POSIX behaviour exists and is not going away; it's a bad move on GNU's part to have defined a symbol with a different prototype and a different behaviour from the POSIX one.

If I don't want the POSIX behaviour, I don't use `basename()`. I think that every project should do the same. I suggest that you go with `gnu_basename()`, and keep patching kmod because using `basename()` is suboptimal instead of keeping patching kmod because it breaks on some systems.

Reply:
My proposal is we do this: https://github.com/kmod-project/kmod/commits/tip-basename/

877280d fixup! fixup! Use portable implementation for basename API
3325c9d fixup! Use portable implementation for basename API
81580c1 Use portable implementation for basename API

I think we can just squash my commits onto yours if you agree.

Reply:
I should probably split the missing.h, but that can be done on top.

Reply:
That will work as long as the project *never* includes `libgen.h`. Which, true, it doesn't have a reason to.

Reply:
@lucasdemarchi are you happy with this patch or do you have comments that should be addressed.

Reply:
Yep, applying it now. Thanks